### PR TITLE
[PF-127] Give RBS more folder level permissions and add template to enable RBS SA use an billing account

### PR DESCRIPTION
**Pull request by yonghaoy**

### CODE REVIEW

File: rbs/variables.tf [hunk 1]
```diff
@@ -103,3 +103,20 @@ locals {
...
+#
+# We separate whether the roles/billing.user is set on the RBS SA as its own flag because the
+# Broad SA that runs terraform does not want to have the broad permissions to set modify
+# permissions on the service account. This was done manually instead for the Broad.
+variable "billing_account_id" {
+  type        = string
+  description = "What billing account to assign to the project."
+}
+
+variable "enable_billing_user" {
```
Comment (wchamber) [inline rbs/variables.tf]: can we merge these variables by using a `default = null` on `billing_account_id` and explain that a null billing account id means we won't set the RBS SA as a billing user?
ditto on terra-env

File: rbs/sa.tf [hunk 2]
```diff
@@ -67,4 +65,10 @@ resource "google_folder_iam_member" "app_folder_roles" {
   member   = "serviceAccount:${google_service_account.app[0].email}"
 }
 
-
+# Grant Terra RBS App Service Account permission use the billing account.
+resource "google_billing_account_iam_member" "app_billing_roles" {
+  count              = var.enable_billing_user ? 1 : 0
```
Comment (wchamber) [inline rbs/sa.tf]: this should also check `var.enable`

**Review by wchamber — CHANGES_REQUESTED** [review]

File: rbs/variables.tf (hunk 1 above)
Comment (yonghaoy) [reply rbs/variables.tf]: Done. Also, I think null doesn't work well, I need to use empty string here. That is my finding in my previous PR to set the optional folder_id

File: rbs/sa.tf (hunk 2 above)
Comment (yonghaoy) [reply rbs/sa.tf]: done

**Review by wchamber — APPROVED** [review]

**Pull request merged**